### PR TITLE
[JA] Translate OTP Distribution into Japanese

### PR DESCRIPTION
**Pull request by Tsuyoshi84**
I translated OTP Distribution page.
Please review! @elixirschool/japanese

### CODE REVIEW

**Comment by Tsuyoshi84** [comment]
BTW, I guess most Japanese are spending long holidays, so it's totally ok to start the review after the holidays. Happy hollidays 😀

**Comment by riseshia** [comment]
I will review this TODAY!!! :bow:

File: ja/lessons/advanced/otp-distribution.md [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+
+## ノード間通信
+
+ElixirはErlang VMで動作しますが、これはErlangの強力な [分散機能](http://erlang.org/doc/reference_manual/distributed.html) へアクセス可能であることを意味します。
+
+> 分散されたErlangシステムは互いに通信するいくつものErlangランタイムシステムを含んでいます。
+これらのそれぞれのランタイムはノードと呼ばれます。
+
+ノードは名前が与えられた任意のErlangランタイムシステムです。
+`iex` セッションを起動して名前を与えることで、ノードを開始できます:
```
Comment (marocchino) [inline ja/lessons/advanced/otp-distribution.md]: maybe)
```suggestion
`iex` セッションを起動してノードを開始しながら名前を与えることができます:
```

File: ja/lessons/advanced/otp-distribution.md [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+別のターミナルウィンドウで、もう一つのノードを起動してみましょう:
+
+```bash
+iex --sname kate@localhost
+iex(kate@localhost)>
+```
+
+これら2つのノードは `Node.spawn_link/1` を使うことでもう一方にメッセージを送ることができます。
+
+### `Node.spawn_link/1` による通信
```
Comment (marocchino) [inline ja/lessons/advanced/otp-distribution.md]: isn't it `Node.spawn_link/2`? https://hexdocs.pm/elixir/Node.html#spawn_link/2

File: ja/lessons/advanced/otp-distribution.md (hunk 1 above)
Comment (marocchino) [reply ja/lessons/advanced/otp-distribution.md]: I mean 'name it' part is more important here.

File: ja/lessons/advanced/otp-distribution.md [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+
+この関数は2つの引数を取ります:
+* 接続したいノードの名前
+* そのノードで動作しているリモートプロセスによって実行させる関数
+
+これはリモートノードとの間で通信を確立して、ノード上で与えられた関数を実行し、リンクされたプロセスのPIDを返します。
+
+`Kate` というモジュールを、その紹介方法を知っている `kate` ノードの中に定義しましょう:
```
Comment (marocchino) [inline ja/lessons/advanced/otp-distribution.md]: maybe)
```suggestion
Kateさんを紹介する `Kate` というモジュールを、 `kate` ノードの中に定義しましょう:
```

File: ja/lessons/advanced/otp-distribution.md [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+これは、ローカルノードが **グループリーダー** であるためです。
+Erlang VMはプロセスを通してI/Oを管理します。
+これによって `IO.puts` のようなI/Oタスクを分散されたノード間で実行することができます。
+これらの分散されたプロセスは、I/Oプロセスグループリーダーによって管理されます。
+このグループリーダーは常にプロセスを生産するノードとなります。
+そのため、 `alex` ノードは `spawn_link/2` から呼び出されているので、そのノードがグループリーダーとなり、 `IO.puts` の出力はそのノードの標準出力ストリームへと向けられます。
+
+#### メッセージへの返信
+
+受信したノードから送信側に対して何らかの *返信* を返したい場合はどうしましょう？ これは、単に `receive/1` と [`send/2`](https://hexdocs.pm/elixir/Process.html#send/3) を使うことで実現できます。
```
Comment (marocchino) [inline ja/lessons/advanced/otp-distribution.md]: Oh.. send/2 in caption and send/3 in a link.

File: ja/lessons/advanced/otp-distribution.md [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+iex(kate@localhost)>
+```
+
+すると `alex` ノードの `iex` セッションでこのメッセージが表示されます:
+
+```elixir
+iex(alex@localhost)> how are you?
+```
+
+コードに戻って何が起こっているのかをブレークダウンしてみましょう。
```
Comment (marocchino) [inline ja/lessons/advanced/otp-distribution.md]:
```suggestion
手を止めて休みながらコードに戻ってみましょう。
```

File: ja/lessons/advanced/otp-distribution.md (hunk 2 above)
Comment (riseshia) [reply ja/lessons/advanced/otp-distribution.md]: It seems original text is wrong..
https://github.com/elixirschool/elixirschool/pull/1841

File: ja/lessons/advanced/otp-distribution.md (hunk 4 above)
Comment (riseshia) [reply ja/lessons/advanced/otp-distribution.md]: original text is wrong... https://github.com/elixirschool/elixirschool/pull/1841

File: ja/lessons/advanced/otp-distribution.md [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
+次のコマンドでアプリを生成します:
+
+```
+mix new chat --sup
+```
+
+### スーパービジョンツリーへのタスクスーパーバイザの追加
+
+タスクスーパーバイザはタスクを動的に監視します。
+これは子が無い状態で開始され、大抵は自身の監視の _下に_ あり、後に任意の数のタスクの監視に使用することができます。
```
Comment (riseshia) [inline ja/lessons/advanced/otp-distribution.md]: I think `often _under_ a supervisor of its own` means, It often has its supervisor. so,
`自分を監視するスーバーバイザの _下に_ あり` may be a right translation 🤔

File: ja/lessons/advanced/otp-distribution.md [hunk 7]
```diff
@@ -0,0 +1,486 @@
...
+      {Task.Supervisor, name: Chat.TaskSupervisor}
+    ]
+
+    opts = [strategy: :one_for_one, name: Chat.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+end
+```
+
+これでアプリケーションが指定されたどこのノードで開始されても、 `Chat.Supervisor` が開始されてタスクを監視する準備ができることがわかります。
```
Comment (riseshia) [inline ja/lessons/advanced/otp-distribution.md]: (maybe)
```suggestion
これでアプリケーションが指定されたどこのノードで開始されても、 `Chat.Supervisor` が開始されてタスクを監視する準備ができたのがわかります。
```

File: ja/lessons/advanced/otp-distribution.md (hunk 5 above)
Comment (riseshia) [reply ja/lessons/advanced/otp-distribution.md]: I think `詳しく` is enough. `なにが起きてるのか分解してみましょう` sounds strange, doesn't it? 🤔

File: ja/lessons/advanced/otp-distribution.md [hunk 8]
```diff
@@ -0,0 +1,486 @@
...
+```elixir
+iex(alex@localhost)> how are you?
+```
+
+コードに戻って何が起こっているのかをブレークダウンしてみましょう。
+
+監視されたタスクを実行したいリモートノードの名前と、そのノードに送信したいメッセージを受け取る `Chat.send_message/2` 関数を持っています。
+
+この関数は `spawn_task/4` 関数を呼び出し、指定された名前のリモートノード上で非同期タスクを実行して、 `Chat.TaskSupervisor` によってリモートノードで監視されます。
+そのノードがチャットアプリケーションのインスタンス _も_ 実行していて、 `Chat.TaskSupervisor` がチャットアプリのスーパービジョンツリーの一部として開始されているため、 `Chat.TaskSupervisor` という名前でタスクスーパーバイザがそのノードで実行されていることがわかります。
```
Comment (riseshia) [inline ja/lessons/advanced/otp-distribution.md]: (maybe)
```suggestion
そのノード _も_ チャットアプリケーションのインスタンスを実行しているので、 `Chat.TaskSupervisor` がチャットアプリのスーパービジョンツリーの一部として開始されているため、 `Chat.TaskSupervisor` という名前でタスクスーパーバイザがそのノードで実行されていることがわかります。
```

File: ja/lessons/advanced/otp-distribution.md [hunk 9]
```diff
@@ -0,0 +1,486 @@
...
+`alex` に `moebi` へとメッセージを送らせましょう:
+
+```elixir
+iex(alex@localhost)> Chat.send_message(:moebi@localhost, "hi")
+chicken?
+:ok
+```
+
+`alex` ノードは `"chicken?"` という返信を受け取ったことがわかります。
+`kate` ノードを開始すると、 `alex` も `moebi` も何も送っていないので、メッセージが来ていないことがわかります(ごめんね `kate`)。
```
Comment (riseshia) [inline ja/lessons/advanced/otp-distribution.md]: (maybe)
```suggestion
`kate` ノードを確認すると、 `alex` も `moebi` も何も送っていないので、メッセージが来ていないことがわかります(ごめんね `kate`)。
```

File: ja/lessons/advanced/otp-distribution.md (hunk 1 above)
Comment (riseshia) [reply ja/lessons/advanced/otp-distribution.md]: we can't call a runtime without given name as node, so It seems "名前を与えて `iex` セッションを起動してことで、ノードを開始できます:" more natural, how do you think?

**Comment by Tsuyoshi84** [comment]
Thank you! I'll check them out later!

File: ja/lessons/advanced/otp-distribution.md (hunk 1 above)
Comment (Tsuyoshi84) [reply ja/lessons/advanced/otp-distribution.md]: Makes sense! I'll fix it

File: ja/lessons/advanced/otp-distribution.md (hunk 5 above)
Comment (Tsuyoshi84) [reply ja/lessons/advanced/otp-distribution.md]: Yeah, could be right. 
OK, I'll just change this part to `詳しく`!

File: ja/lessons/advanced/otp-distribution.md (hunk 6 above)
Comment (Tsuyoshi84) [reply ja/lessons/advanced/otp-distribution.md]: I think you're right! 👍

File: ja/lessons/advanced/otp-distribution.md (hunk 1 above)
Comment (riseshia) [reply ja/lessons/advanced/otp-distribution.md]: LGTM to me :)

**Pull request merged**